src/tribler-core/tribler_core/modules/bandwidth_accounting/transaction.py

```diff
@@ -0,0 +1,200 @@
+"""
+This file defines the required data structures for the bandwidth accounting mechanism.
+Note that we define two different class types of BandwidthTransaction: one for the object that resides outside the
+Pony database and another one that represents the class inside Pony.
+We make this separation to workaround the fact that Pony does not support objects that are created outside a database
```

Comment:
As I understand, the motivation for creating two different classes is because of two reasons:
1) Pony does not allow creating objects with `Required` fields which are not filled yet, and there were some places in the code where it is desirable to create a partially filled object (is it still so? @devos50, can you point to places in the code where it is desirable to have partially filled objects?)
2) Pony currently does not allow to have "detached" objects, which are created but not stored in the database as fast as possible. Like, if you want to have some long-lived in-memory object not stored in the database and not tied to a particular db_session.

If these reasons are still valid, I think it is normal to create such objects. The code repetition is actually minimal, and the methods are pretty different.

But it may be confusing that the two classes have the same name, `BandwidthTransaction`. How about name non-ORM class something like `BandwidthTransactionData`?

Also, it is possible to define the non-ORM class as a `@dataclass` to remove some boilerplate code:
https://docs.python.org/3.8/library/dataclasses.html

Reply:
Thanks for the `@dataclass` decorator! I didn't know about that :)